### PR TITLE
HTML: The "rendered legend" creates a block formatting context

### PR DESCRIPTION
**Pull request by zcorpan**
Follows https://github.com/whatwg/html/pull/2718

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/238269373)

# Firefox (nightly) #
Testing web-platform-tests at revision 36c5b560d73ec367f0a968a136a11159901d4264
Using browser at version BuildID 20170531100318; SourceStamp 7b8937970f9ca85db88cb2496f2112175fd847c8
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6125/html/rendering/non-replaced-elements/the-fieldset-element-0/legend-block-formatting-context.html">/html/rendering/non-replaced-elements/the-fieldset-element-0/legend-block-formatting-context.html</a></summary>


|          Subtest           | Results |                          Messages                         |
|----------------------------|---------|-----------------------------------------------------------|
|                            | OK      |                                                           |
| `in-body`                  | PASS    |                                                           |
| `rendered-legend`          | PASS    |                                                           |
| `in-fieldset-second-child` | FAIL    | `assert_equals: legend.offsetLeft expected 2 but got 52`  |
| `in-fieldset-descendant`   | FAIL    | `assert_equals: legend.offsetTop expected 50 but got 100` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/6125/html/rendering/non-replaced-elements/the-fieldset-element-0/legend.html">/html/rendering/non-replaced-elements/the-fieldset-element-0/legend.html</a></summary>


|                  Subtest                  | Results |                     Messages                    |
|-------------------------------------------|---------|-------------------------------------------------|
|                                           | OK      |                                                 |
| `in-body: display`                        | PASS    |                                                 |
| `in-body: unicodeBidi`                    | PASS    |                                                 |
| `in-body: marginTop`                      | PASS    |                                                 |
| `in-body: marginRight`                    | PASS    |                                                 |
| `in-body: marginBottom`                   | PASS    |                                                 |
| `in-body: marginLeft`                     | PASS    |                                                 |
| `in-body: paddingTop`                     | PASS    |                                                 |
| `in-body: paddingRight`                   | FAIL    | `assert_equals: expected "2px" but got "0px"`   |
| `in-body: paddingBottom`                  | PASS    |                                                 |
| `in-body: paddingLeft`                    | FAIL    | `assert_equals: expected "2px" but got "0px"`   |
| `in-body: overflow`                       | PASS    |                                                 |
| `in-body: height`                         | PASS    |                                                 |
| `in-body: box-sizing`                     | PASS    |                                                 |
| `in-body: width`                          | PASS    |                                                 |
| `rendered-legend: display`                | PASS    |                                                 |
| `rendered-legend: unicodeBidi`            | PASS    |                                                 |
| `rendered-legend: marginTop`              | PASS    |                                                 |
| `rendered-legend: marginRight`            | PASS    |                                                 |
| `rendered-legend: marginBottom`           | PASS    |                                                 |
| `rendered-legend: marginLeft`             | PASS    |                                                 |
| `rendered-legend: paddingTop`             | PASS    |                                                 |
| `rendered-legend: paddingRight`           | PASS    |                                                 |
| `rendered-legend: paddingBottom`          | PASS    |                                                 |
| `rendered-legend: paddingLeft`            | PASS    |                                                 |
| `rendered-legend: overflow`               | PASS    |                                                 |
| `rendered-legend: height`                 | PASS    |                                                 |
| `rendered-legend: box-sizing`             | PASS    |                                                 |
| `rendered-legend: width`                  | PASS    |                                                 |
| `in-fieldset-second-child: display`       | PASS    |                                                 |
| `in-fieldset-second-child: unicodeBidi`   | PASS    |                                                 |
| `in-fieldset-second-child: marginTop`     | PASS    |                                                 |
| `in-fieldset-second-child: marginRight`   | PASS    |                                                 |
| `in-fieldset-second-child: marginBottom`  | PASS    |                                                 |
| `in-fieldset-second-child: marginLeft`    | PASS    |                                                 |
| `in-fieldset-second-child: paddingTop`    | PASS    |                                                 |
| `in-fieldset-second-child: paddingRight`  | PASS    |                                                 |
| `in-fieldset-second-child: paddingBottom` | PASS    |                                                 |
| `in-fieldset-second-child: paddingLeft`   | PASS    |                                                 |
| `in-fieldset-second-child: overflow`      | PASS    |                                                 |
| `in-fieldset-second-child: height`        | PASS    |                                                 |
| `in-fieldset-second-child: box-sizing`    | PASS    |                                                 |
| `in-fieldset-second-child: width`         | FAIL    | `assert_not_equals: got disallowed value "0px"` |
| `in-fieldset-descendant: display`         | PASS    |                                                 |
| `in-fieldset-descendant: unicodeBidi`     | PASS    |                                                 |
| `in-fieldset-descendant: marginTop`       | PASS    |                                                 |
| `in-fieldset-descendant: marginRight`     | PASS    |                                                 |
| `in-fieldset-descendant: marginBottom`    | PASS    |                                                 |
| `in-fieldset-descendant: marginLeft`      | PASS    |                                                 |
| `in-fieldset-descendant: paddingTop`      | PASS    |                                                 |
| `in-fieldset-descendant: paddingRight`    | FAIL    | `assert_equals: expected "2px" but got "0px"`   |
| `in-fieldset-descendant: paddingBottom`   | PASS    |                                                 |
| `in-fieldset-descendant: paddingLeft`     | FAIL    | `assert_equals: expected "2px" but got "0px"`   |
| `in-fieldset-descendant: overflow`        | PASS    |                                                 |
| `in-fieldset-descendant: height`          | PASS    |                                                 |
| `in-fieldset-descendant: box-sizing`      | PASS    |                                                 |
| `in-fieldset-descendant: width`           | PASS    |                                                 |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/238269374)

# Sauce (safari) #
Testing web-platform-tests at revision 36c5b560d73ec367f0a968a136a11159901d4264
Using browser at version 10.0
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6125/html/rendering/non-replaced-elements/the-fieldset-element-0/legend-block-formatting-context.html">/html/rendering/non-replaced-elements/the-fieldset-element-0/legend-block-formatting-context.html</a></summary>


|          Subtest           | Results |                          Messages                         |
|----------------------------|---------|-----------------------------------------------------------|
|                            | OK      |                                                           |
| `in-body`                  | FAIL    | `assert_equals: legend.offsetLeft expected 0 but got 50`  |
| `rendered-legend`          | FAIL    | `assert_equals: legend.offsetTop expected 0 but got 1`    |
| `in-fieldset-second-child` | FAIL    | `assert_equals: legend.offsetLeft expected 2 but got 52`  |
| `in-fieldset-descendant`   | FAIL    | `assert_equals: legend.offsetLeft expected 2 but got 104` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/6125/html/rendering/non-replaced-elements/the-fieldset-element-0/legend.html">/html/rendering/non-replaced-elements/the-fieldset-element-0/legend.html</a></summary>


|                  Subtest                  | Results |                     Messages                    |
|-------------------------------------------|---------|-------------------------------------------------|
|                                           | OK      |                                                 |
| `in-body: display`                        | PASS    |                                                 |
| `in-body: unicodeBidi`                    | PASS    |                                                 |
| `in-body: marginTop`                      | PASS    |                                                 |
| `in-body: marginRight`                    | PASS    |                                                 |
| `in-body: marginBottom`                   | PASS    |                                                 |
| `in-body: marginLeft`                     | PASS    |                                                 |
| `in-body: paddingTop`                     | PASS    |                                                 |
| `in-body: paddingRight`                   | PASS    |                                                 |
| `in-body: paddingBottom`                  | PASS    |                                                 |
| `in-body: paddingLeft`                    | PASS    |                                                 |
| `in-body: overflow`                       | PASS    |                                                 |
| `in-body: height`                         | PASS    |                                                 |
| `in-body: box-sizing`                     | PASS    |                                                 |
| `in-body: width`                          | FAIL    | `assert_not_equals: got disallowed value "0px"` |
| `rendered-legend: display`                | PASS    |                                                 |
| `rendered-legend: unicodeBidi`            | PASS    |                                                 |
| `rendered-legend: marginTop`              | PASS    |                                                 |
| `rendered-legend: marginRight`            | PASS    |                                                 |
| `rendered-legend: marginBottom`           | PASS    |                                                 |
| `rendered-legend: marginLeft`             | PASS    |                                                 |
| `rendered-legend: paddingTop`             | PASS    |                                                 |
| `rendered-legend: paddingRight`           | PASS    |                                                 |
| `rendered-legend: paddingBottom`          | PASS    |                                                 |
| `rendered-legend: paddingLeft`            | PASS    |                                                 |
| `rendered-legend: overflow`               | PASS    |                                                 |
| `rendered-legend: height`                 | PASS    |                                                 |
| `rendered-legend: box-sizing`             | PASS    |                                                 |
| `rendered-legend: width`                  | PASS    |                                                 |
| `in-fieldset-second-child: display`       | PASS    |                                                 |
| `in-fieldset-second-child: unicodeBidi`   | PASS    |                                                 |
| `in-fieldset-second-child: marginTop`     | PASS    |                                                 |
| `in-fieldset-second-child: marginRight`   | PASS    |                                                 |
| `in-fieldset-second-child: marginBottom`  | PASS    |                                                 |
| `in-fieldset-second-child: marginLeft`    | PASS    |                                                 |
| `in-fieldset-second-child: paddingTop`    | PASS    |                                                 |
| `in-fieldset-second-child: paddingRight`  | PASS    |                                                 |
| `in-fieldset-second-child: paddingBottom` | PASS    |                                                 |
| `in-fieldset-second-child: paddingLeft`   | PASS    |                                                 |
| `in-fieldset-second-child: overflow`      | PASS    |                                                 |
| `in-fieldset-second-child: height`        | PASS    |                                                 |
| `in-fieldset-second-child: box-sizing`    | PASS    |                                                 |
| `in-fieldset-second-child: width`         | FAIL    | `assert_not_equals: got disallowed value "0px"` |
| `in-fieldset-descendant: display`         | PASS    |                                                 |
| `in-fieldset-descendant: unicodeBidi`     | PASS    |                                                 |
| `in-fieldset-descendant: marginTop`       | PASS    |                                                 |
| `in-fieldset-descendant: marginRight`     | PASS    |                                                 |
| `in-fieldset-descendant: marginBottom`    | PASS    |                                                 |
| `in-fieldset-descendant: marginLeft`      | PASS    |                                                 |
| `in-fieldset-descendant: paddingTop`      | PASS    |                                                 |
| `in-fieldset-descendant: paddingRight`    | PASS    |                                                 |
| `in-fieldset-descendant: paddingBottom`   | PASS    |                                                 |
| `in-fieldset-descendant: paddingLeft`     | PASS    |                                                 |
| `in-fieldset-descendant: overflow`        | PASS    |                                                 |
| `in-fieldset-descendant: height`          | PASS    |                                                 |
| `in-fieldset-descendant: box-sizing`      | PASS    |                                                 |
| `in-fieldset-descendant: width`           | FAIL    | `assert_not_equals: got disallowed value "0px"` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/238269380)

# Chrome (unstable) #
Testing web-platform-tests at revision 36c5b560d73ec367f0a968a136a11159901d4264
Using browser at version 60.0.3112.7 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6125/html/rendering/non-replaced-elements/the-fieldset-element-0/legend-block-formatting-context.html">/html/rendering/non-replaced-elements/the-fieldset-element-0/legend-block-formatting-context.html</a></summary>


|          Subtest           | Results |                          Messages                         |
|----------------------------|---------|-----------------------------------------------------------|
|                            | OK      |                                                           |
| `in-body`                  | FAIL    | `assert_equals: legend.offsetLeft expected 0 but got 50`  |
| `rendered-legend`          | FAIL    | `assert_equals: legend.offsetTop expected 0 but got 1`    |
| `in-fieldset-second-child` | FAIL    | `assert_equals: legend.offsetLeft expected 2 but got 52`  |
| `in-fieldset-descendant`   | FAIL    | `assert_equals: legend.offsetLeft expected 2 but got 104` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/6125/html/rendering/non-replaced-elements/the-fieldset-element-0/legend.html">/html/rendering/non-replaced-elements/the-fieldset-element-0/legend.html</a></summary>


|                  Subtest                  | Results |                       Messages                       |
|-------------------------------------------|---------|------------------------------------------------------|
|                                           | OK      |                                                      |
| `in-body: display`                        | PASS    |                                                      |
| `in-body: unicodeBidi`                    | FAIL    | `assert_equals: expected "isolate" but got "normal"` |
| `in-body: marginTop`                      | PASS    |                                                      |
| `in-body: marginRight`                    | PASS    |                                                      |
| `in-body: marginBottom`                   | PASS    |                                                      |
| `in-body: marginLeft`                     | PASS    |                                                      |
| `in-body: paddingTop`                     | PASS    |                                                      |
| `in-body: paddingRight`                   | PASS    |                                                      |
| `in-body: paddingBottom`                  | PASS    |                                                      |
| `in-body: paddingLeft`                    | PASS    |                                                      |
| `in-body: overflow`                       | PASS    |                                                      |
| `in-body: height`                         | PASS    |                                                      |
| `in-body: box-sizing`                     | PASS    |                                                      |
| `in-body: width`                          | FAIL    | `assert_not_equals: got disallowed value "0px"`      |
| `rendered-legend: display`                | PASS    |                                                      |
| `rendered-legend: unicodeBidi`            | FAIL    | `assert_equals: expected "isolate" but got "normal"` |
| `rendered-legend: marginTop`              | PASS    |                                                      |
| `rendered-legend: marginRight`            | PASS    |                                                      |
| `rendered-legend: marginBottom`           | PASS    |                                                      |
| `rendered-legend: marginLeft`             | PASS    |                                                      |
| `rendered-legend: paddingTop`             | PASS    |                                                      |
| `rendered-legend: paddingRight`           | PASS    |                                                      |
| `rendered-legend: paddingBottom`          | PASS    |                                                      |
| `rendered-legend: paddingLeft`            | PASS    |                                                      |
| `rendered-legend: overflow`               | PASS    |                                                      |
| `rendered-legend: height`                 | PASS    |                                                      |
| `rendered-legend: box-sizing`             | PASS    |                                                      |
| `rendered-legend: width`                  | PASS    |                                                      |
| `in-fieldset-second-child: display`       | PASS    |                                                      |
| `in-fieldset-second-child: unicodeBidi`   | FAIL    | `assert_equals: expected "isolate" but got "normal"` |
| `in-fieldset-second-child: marginTop`     | PASS    |                                                      |
| `in-fieldset-second-child: marginRight`   | PASS    |                                                      |
| `in-fieldset-second-child: marginBottom`  | PASS    |                                                      |
| `in-fieldset-second-child: marginLeft`    | PASS    |                                                      |
| `in-fieldset-second-child: paddingTop`    | PASS    |                                                      |
| `in-fieldset-second-child: paddingRight`  | PASS    |                                                      |
| `in-fieldset-second-child: paddingBottom` | PASS    |                                                      |
| `in-fieldset-second-child: paddingLeft`   | PASS    |                                                      |
| `in-fieldset-second-child: overflow`      | PASS    |                                                      |
| `in-fieldset-second-child: height`        | PASS    |                                                      |
| `in-fieldset-second-child: box-sizing`    | PASS    |                                                      |
| `in-fieldset-second-child: width`         | FAIL    | `assert_not_equals: got disallowed value "0px"`      |
| `in-fieldset-descendant: display`         | PASS    |                                                      |
| `in-fieldset-descendant: unicodeBidi`     | FAIL    | `assert_equals: expected "isolate" but got "normal"` |
| `in-fieldset-descendant: marginTop`       | PASS    |                                                      |
| `in-fieldset-descendant: marginRight`     | PASS    |                                                      |
| `in-fieldset-descendant: marginBottom`    | PASS    |                                                      |
| `in-fieldset-descendant: marginLeft`      | PASS    |                                                      |
| `in-fieldset-descendant: paddingTop`      | PASS    |                                                      |
| `in-fieldset-descendant: paddingRight`    | PASS    |                                                      |
| `in-fieldset-descendant: paddingBottom`   | PASS    |                                                      |
| `in-fieldset-descendant: paddingLeft`     | PASS    |                                                      |
| `in-fieldset-descendant: overflow`        | PASS    |                                                      |
| `in-fieldset-descendant: height`          | PASS    |                                                      |
| `in-fieldset-descendant: box-sizing`      | PASS    |                                                      |
| `in-fieldset-descendant: width`           | FAIL    | `assert_not_equals: got disallowed value "0px"`      |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/238269381)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision 36c5b560d73ec367f0a968a136a11159901d4264

File: html/rendering/non-replaced-elements/the-fieldset-element-0/legend-block-formatting-context.html [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+<div class=wrapper>
+ <div class=float></div>
+ <legend id=in-body><div class=float></div></legend>
+ <div class=float></div>
+</div>
+
+<div class=wrapper>
+ <fieldset>
+  <div class=float></div>
```
Comment (mstensho) [inline html/rendering/non-replaced-elements/the-fieldset-element-0/legend-block-formatting-context.html]: https://html.spec.whatwg.org/multipage/forms.html#the-legend-element says that LEGEND needs to be the first child of the FIELDSET. Will #rendered-legend really become a rendered legend here?

Comment (zcorpan) [inline html/rendering/non-replaced-elements/the-fieldset-element-0/legend-block-formatting-context.html]: Yes. https://html.spec.whatwg.org/#rendered-legend says what is the rendered legend. What you cite is an authoring requirement.

Comment (mstensho) [inline html/rendering/non-replaced-elements/the-fieldset-element-0/legend-block-formatting-context.html]: OK, I suppose I forgot to read the entire spec backwards again. :)

**Review by mstensho — APPROVED** [review]

**Pull request merged**